File: akka-samples/scala-samples/src/main/scala/actors/simple/SimplePrinter.scala
package actors.simple

import akka.actor.typed.scaladsl.Behaviors
import akka.actor.typed.{ActorSystem, Behavior}
import actors.simple.Printer.PrintMessage

object SimplePrinter {
  def main(args: Array[String]): Unit = {
    val system = ActorSystem[PrintMessage](Printer(), "simple-printer")
    system ! PrintMessage("Hello there!")
  }
}

private object Printer {
  final case class PrintMessage(text: String)

  def apply(): Behavior[PrintMessage] =
    Behaviors.receive[PrintMessage] { (context, message) =>
      println(message.text)
      Behaviors.same
    }
}

File: akka-samples/java-samples/src/main/java/actors/simple/SimplePrinter.java
package actors.simple;

import akka.actor.typed.ActorSystem;
import akka.actor.typed.Behavior;
import akka.actor.typed.javadsl.AbstractBehavior;
import akka.actor.typed.javadsl.ActorContext;
import akka.actor.typed.javadsl.Behaviors;
import akka.actor.typed.javadsl.Receive;

class SimplePrinter extends AbstractBehavior<SimplePrinter.PrintMessage> {
    static final class PrintMessage {
        public final String text;

        public PrintMessage(String text) {
            this.text = text;
        }
    }

    @Override
    public Receive<PrintMessage> createReceive() {
        return newReceiveBuilder().onMessage(PrintMessage.class, (message) -> {
            System.out.println(message.text);
            return Behaviors.same();
        }).build();
    }

    public SimplePrinter(ActorContext<PrintMessage> context) {
        super(context);
    }

    public static Behavior<PrintMessage> create() {
        return Behaviors.setup(SimplePrinter::new);
    }

    public static void main(String[] args) {
        final ActorSystem<PrintMessage> system = ActorSystem.create(SimplePrinter.create(), "simple-printer");
        system.tell(new PrintMessage("Hello there!"));
    }
}